### PR TITLE
✨ feat(types): add support for types on this repository

### LINKED ISSUE #285: Add Typescript Types
It would be great to have Typescript types along with this package.

### PR DESCRIPTION
**Pull request by dominguezcelada**
## 📝 Summary

- Add support for types on this repository

## ☑️ TODO
- [x] Tests
- [x] Try it on playground with `npm link`

## ⛱ Motivation and Context

Fixes #285

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { GraphQLError } from "graphql";
+
+type JsonSchema = {
+  [key: string]: any;
```
Comment (dominguezcelada) [inline index.d.ts:4]: We are importing a plain `json` file I'm assuming this level of typing is enough. Am I right?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+type Schema = {
+  idl: string;
+  json: JsonSchema;
+};
+
+declare module "@octokit/graphql-schema" {
+  function validate(
+    query: ReadonlyArray<string> | Readonly<string>
```
Comment (dominguezcelada) [inline index.d.ts:14]: Reference: https://github.com/apollographql/graphql-tag/blob/9b719a77530b81acc48be238e2dee3cf28e65beb/index.d.ts#L3

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  ): ReadonlyArray<GraphQLError>;
```
Comment (dominguezcelada) [inline index.d.ts:15]: Reference:
https://github.com/graphql/graphql-js/blob/9f800d236b08c8344fe453d28b2bd3add38ce812/src/type/validate.d.ts#L12-L14

**Comment by gr2m** [comment]
@blimmer could you check if that PR resolves your problem from #285?

File: index.d.ts (hunk 1 above)
Comment (blimmer) [reply index.d.ts:4]: Yeah, I'd think this is fine!

**Pull request merged**